homeassistant/components/plant/__init__.py

```diff
@@ -357,6 +371,20 @@ def extra_state_attributes(self):
         if self._brightness_history.max is not None:
             attrib[ATTR_MAX_BRIGHTNESS_HISTORY] = self._brightness_history.max
 
+        attributes = (
```

Comment:
These add static attributes that are stored in our state machine (and thus add memory usage and cost storage space in database history recording), while their only use is a custom UI card.

I suggest on making this configurable on the card configuration instead in such cases.

Reply:
I suggested to expose theses values to attributes like the `mired_min/mired_max` for light to avoid the user to enter the value twice (in the integration and in the card).
Also, there is default value and user has no idea about these default value unless he go to the documentation.

But I understand your arguments about database performance for static information.

I will look how we can deal with it in lovelace card. Adding 8 input number looks a little overloaded 😅

Reply:
> I suggested to expose theses attributes values like the mired_min/mired_max for light for example (or the min max values for climate entity) 

That is not comparable, as those can be provided by an integration (e.g., the Zigbee network or whatever else).

This is static manual configuration.

Reply:
Also, I don't know the status of this plant entity in the HA core code base. It's `yaml` only, there is few commits on it and it looks like a helper but very specific. Maybe it can be migrated to a sensor helper like the utility meter or it's better to use template sensor for these rules.

Reply:
> Also, I don't know the status of this plant entity in the HA core codebase.

I don't know either. There is no state at this point. As a helper it might work; however, in the end, I think this platform, in general, isn't providing anything really.

For example, its state is "problem" or "OK", ergo: A binary state. It should be a binary sensor entity, not a `plant` entity. We have a couple of more of these occurrences (e.g., promixity; which should be just a sensor actually).

I think in the long-term, the chances that we convert these into their proper entity types. As for `plant` itself, I'm not sure if a dedicated entity would be even warranted. It is merely used as a frontend thing and for automations using all original sensor entities, in general, would be easier to automate with/on (including their min/max values).

Reply:
Totally agree 👍 

Note that `utility_meter` sensors expose config values but it's just to tease you 🤪

e.g.
```yaml
meter_period: daily
cron pattern: 0 0 * * *
```

Reply:
> Note that utility_meter sensors expose config values but it's just to tease you

More do stuff like that, however, we do try to minimize it nowadays. Things from the past we did a lot and learned from 😉

Reply:
> 

That's not the only use though. I have wanted to have those states available for automation scopes (mainly notifications) - not to use it for a custom UI card.

Reply:
> I have wanted to have those states available for automation scopes (mainly notifications), so would appreciate to be allowed for!

The point made above, is that those are not states; they are configurations.

Reply:
> 
> The point made above, is that those are not states; they are configurations.

Sorry, yes I understand. Point however is that it is useful to have these configurations available somehow to use in automations in addition to the UI cards.

Reply:
Sure, but they are not state, thus don't belong in the state engine 🤷